packages/doc/src/MarkdownCommentStore.ts

```diff
@@ -0,0 +1,148 @@
+import { readFile } from "fs/promises";
+import { EntityDbMetadata } from "joist-codegen";
+import * as path from "path";
+import { remark } from "remark";
+import { CommentStore, EnumData, FieldSourceType } from "./CommentStore";
+import { hashFile } from "./utils";
+
+/**
+ * A Markdown CommentStore with opinionated but loose semantics around
+ * markdown document structure
+ */
+export class MarkdownCommentStore extends CommentStore {
+  private documents = new Map<string, Promise<null | ReturnType<(typeof remark)["parse"]>>>();
+
+  async forField(entity: EntityDbMetadata, fieldName: string, source: FieldSourceType, generated: boolean) {
+    return this.withGenerated(entity.name, generated, await this.findHeadingContent(entity.name, fieldName));
+  }
+
+  async forEntity(entity: EntityDbMetadata, generated: boolean): Promise<string | void> {
```

Comment:
Huh, I've not used `| void` as an empty return value for; can we use `| undefined` instead? Just is what I'm more used to.